tests/telemetry/test_writer.py

```diff
@@ -289,7 +289,7 @@ def test_app_started_event_configuration_override(test_agent_session, run_python
         {"name": "DD_API_SECURITY_SAMPLE_DELAY", "origin": "default", "value": 30.0},
         {"name": "DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING", "origin": "default", "value": ""},
         {"name": "DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING_ENABLED", "origin": "default", "value": True},
-        {"name": "DD_APPSEC_AUTO_USER_INSTRUMENTATION_MODE", "origin": "default", "value": ""},
+        {"name": "DD_APPSEC_AUTO_USER_INSTRUMENTATION_MODE", "origin": "unknown", "value": "identification"},
```

Comment:
We are now captured the parsed value of `DD_APPSEC_AUTO_USER_INSTRUMENTATION_MODE`. This parsed value is not equal to the default (empty string) and it's not set by environment variables. Therefore the source is "unknown". This is not ideal but fixing this issue is beyond the scope of this PR.

Reply:
I think the "problem" is here https://github.com/DataDog/dd-trace-py/blob/dacfb8d3d202a8a187caf7c3dd2cd813f77c41a1/ddtrace/settings/asm.py#L219-L221, or here https://github.com/DataDog/dd-trace-py/blob/dacfb8d3d202a8a187caf7c3dd2cd813f77c41a1/ddtrace/settings/asm.py#L105